src/Twig/Runtime/CompatRuntime.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Bolt\Twig\Runtime;
+
+use Bolt\Config;
+use Bolt\Storage\Collection\Taxonomy;
+
+/**
+ * Twig helpers that smooth the transition from Legacy to new Storage layers.
+ *
+ * @internal
+ */
+class CompatRuntime
+{
+    /** @var Config $appConfig */
+    private $appConfig;
+
+    /**
+     * Constructor.
+     *
+     * @param Config $appConfig
+     */
+    public function __construct($appConfig)
+    {
+        $this->appConfig = $appConfig;
+    }
+
+    /**
+     *
+     *
+     * @param array|Taxonomy $taxonomy
+     * @return array
+     *
+     */
+    public function taxonomyCompat($taxonomy)
+    {
+        if ($this->isLegacy()) {
+            return $taxonomy;
+        }
+
+        $compiled = [];
+        foreach ($taxonomy as $el) {
+            $compiled[$el->getTaxonomytype()]['/' . $el->getTaxonomytype() . '/' . $el->getSlug()] = $el->getSlug();
+        }
+
+        return $compiled;
+    }
+
+    protected function isLegacy()
+    {
+        return $this->appConfig->get('general/compatibility/setcontent_legacy', true);
+    }
```

Comment:
Wouldn't it cover more bases to drop this and instead check the type in `taxonomyCompat()`